python/tskit/trees.py

```diff
@@ -1403,12 +1403,23 @@ def is_descendant(self, u, v):
     def num_nodes(self):
         """
         Returns the number of nodes in the :class:`TreeSequence` this tree is in.
-        Equivalent to ``tree.tree_sequence.num_nodes``. To find the number of
-        nodes that are reachable from all roots use ``len(list(tree.nodes()))``.
+        Equivalent to ``tree.tree_sequence.num_nodes``.
+
+        .. deprecated:: 0.4
+            Use :attr:`Tree.tree_sequence.num_nodes<TreeSequence.num_nodes>` if you want
+            the number of nodes in the entire tree sequence, or
+            ``len(list(tree.nodes()))`` to find the number of nodes that are
```

Comment:
I guess we should make this as efficient as we can, so ``len(tree.preorder())`` would be better.

Reply:
Yes, nice one